tailscale/rootfs/etc/s6-overlay/s6-rc.d/post-tailscaled/run

```diff
@@ -44,22 +44,70 @@ do
   sleep 1
 done
 
+# Wait for the network to be available
+while ! /opt/tailscale status --json --peers=false --self=false \
+  | jq --exit-status '.BackendState == "Running" or .BackendState == "NeedsLogin"' > /dev/null;
+do
+  sleep 2
+done
+
+# Start Tailscale
+IFS=","
+if ! /opt/tailscale up \
+  --hostname "$(bashio::info.hostname)" \
+  --advertise-exit-node \
+  --accept-routes \
+  --advertise-routes="${routes[*]}" \
+  --advertise-tags="${tags}";
+then
+  bashio::log.error "Unable to start up Tailscale"
+  bashio::exit.nok
+fi
+
 # Wait for the network to be available and logged in
-while true;
+while ! /opt/tailscale status --json --peers=false --self=false \
+  | jq --exit-status '.BackendState == "Running"' > /dev/null
 do
-  if /opt/tailscale status --json --peers=false --self=false \
-    | jq --exit-status '.BackendState == "Running" or .BackendState == "NeedsLogin"' > /dev/null;
-  then
-    IFS=","
-    /opt/tailscale up \
-      --hostname "$(bashio::info.hostname)" \
-      --advertise-exit-node \
-      --accept-routes \
-      --advertise-routes="${routes[*]}" \
-      --advertise-tags="${tags}"
-
-    bashio::exit.ok
-  fi
-  # Well... wait a bit more
   sleep 2
 done
+
+bashio::log.info "Tailscale is running"
+bashio::log.info "Tailscale version: $(/opt/tailscale version --daemon=false | head -n 1)"
+
+bashio::log.info "Enabling Tailscale Proxy..."
+
+declare domain
+
+# Remove previously configured proxy first, because the configuration is persistent between reboots
+/opt/tailscale serve https:443 / "http://localhost:$(bashio::core.port)" off &> /dev/null || true
```

Comment:
IMHO, we should check the current state and make it match and not just turn it off.